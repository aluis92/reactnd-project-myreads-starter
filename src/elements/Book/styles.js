import styled, { css } from 'styled-components';
import arrowDropDown from '../../assets/icons/arrow-drop-down.svg'

const bookFontSize = css`
    font-size: 0.8em;
`;

export const Container = styled.div`
    width: 140px;
`;

export const BookTitle = styled.h3`
    ${bookFontSize}

    margin-top: 10px;
`;

export const BookAuthors = styled.p`
    ${bookFontSize}

    color: #999;
`;

export const BookTop = styled.div`
    position: relative;
    height: 200px;
    display: flex;
    align-items: flex-end;
`;

export const BookCover = styled.div`
    width: 128px;
    height: 193px;
    box-shadow: 0 3px 6px rgba(0,0,0,0.16), 0 3px 6px rgba(0,0,0,0.23);
    background: #eee;
`;
