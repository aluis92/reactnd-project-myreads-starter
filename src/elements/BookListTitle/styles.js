import styled from 'styled-components';

export const Container = styled.header`
    padding: 10px 0;
    background: #2e7c31;
    text-align: center;
`;

export const Title = styled.h1`
    font-weight: 400;
    margin: 0;
    color: white;
`;
