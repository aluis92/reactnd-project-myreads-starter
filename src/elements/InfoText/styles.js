import styled from 'styled-components';

export const Container = styled.header`
    padding: 50px;
`;

export const Text = styled.h1`
    text-align: center;
    font-size: 1.5em;
`;
