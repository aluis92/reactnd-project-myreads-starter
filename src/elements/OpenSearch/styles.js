import styled from 'styled-components';
import { Link } from 'react-router-dom';
import addIcon from '../../assets/icons/add.svg';

export const Container = styled.header`
    position: fixed;
    right: 25px;
    bottom: 25px;
`;

export const OpenSearchLink = styled(Link)`
    display: block;
    width: 50px;
    height: 50px;
    border-radius: 50%;
    background: #2e7d32;
    background-image: url(${addIcon});
    background-repeat: no-repeat;
    background-position: center;
    background-size: 28px;
    box-shadow: 0 3px 6px rgba(0,0,0,0.16), 0 3px 6px rgba(0,0,0,0.23);
    font-size: 0;
`;
