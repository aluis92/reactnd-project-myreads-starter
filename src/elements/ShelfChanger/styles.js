import styled from 'styled-components';
import arrowDropDown from '../../assets/icons/arrow-drop-down.svg'

export const Container = styled.div`
    position: absolute;
    right: 0;
    bottom: -10px;
    width: 40px;
    height: 40px;
    border-radius: 50%;
    background: #60ac5d;
    background-image: url(${arrowDropDown});
    background-repeat: no-repeat;
    background-position: center;
    background-size: 20px;
    box-shadow: 0 3px 6px rgba(0,0,0,0.16), 0 3px 6px rgba(0,0,0,0.23);
`;

export const Select = styled.select`
    width: 100%;
    height: 100%;
    opacity: 0;
    cursor: pointer;
`;
