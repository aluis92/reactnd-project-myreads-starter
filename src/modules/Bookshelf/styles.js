import styled from 'styled-components';
import arrowDropDown from '../../assets/icons/arrow-drop-down.svg'

export const Container = styled.div`
    padding: 0 10px 20px;

    @media (min-width: 600px) {
        .bookshelf {
            padding: 0 20px 40px;
        }
    }
`;

export const BookshelfTitle = styled.h2`
    border-bottom: 1px solid #dedede;
`;

export const BookshelfBooks = styled.div`
    text-align: center;
`;

export const BooksGrid = styled.ol`
    list-style-type: none;
    padding: 0;
    margin: 0;
    display: flex;
    justify-content: center;
    flex-wrap: wrap;

    li {
        padding: 10px 15px;
        text-align: left;
    }
`;
