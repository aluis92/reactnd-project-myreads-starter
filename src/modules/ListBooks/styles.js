import styled from 'styled-components';

export const Container = styled.section`
    position: relative
`;

export const Content = styled.div`
    padding: 0 0 80px;
    flex: 1;
`;
