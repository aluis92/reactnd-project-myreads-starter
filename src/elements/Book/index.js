import React from 'react';
import {
    arrayOf,
    string
} from 'prop-types';
import ShelfChanger from '../ShelfChanger';

import {
    BookAuthors,
    BookCover,
    BookTitle,
    BookTop,
    Container,
} from './styles';

function Book({
    imageLinks,
    authors,
    title
}) {
    return (
        <li>
            <Container className="book">
                <BookTop>
                    <BookCover style={{backgroundImage: `url(${imageLinks.thumbnail})` }} />
                    <ShelfChanger />
                </BookTop>
                <BookTitle>{title}</BookTitle>
                <BookAuthors>{authors.join(', ')}</BookAuthors>
            </Container>
        </li>
    );
}

Book.defaultProps = {
    imageLinks: '',
    authors: [],
    title: '',
};

Book.propTypes = {
    imageLinks: string.isRequired,
    authors: arrayOf(string),
    title: string.isRequired,
};

export default Book;