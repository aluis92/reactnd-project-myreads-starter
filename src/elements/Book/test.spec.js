import React from 'react';
import {
    shallow,
    mount,
} from 'enzyme';

import { BookCondition } from './styles';

import Book from './index';

describe('Book component', () => {
    const defaultBookProps = {
        id: '',
        imageLinks: {},
        authors: [],
        title: '',
        onChangeShelf: jest.fn(),
    };

    it('should render without crashing', () => {
        expect(shallow(
            <Book
                {...defaultBookProps}
            />,
        )).toMatchSnapshot();
    });

    it('should to show added category', () => {
        const bookProps = {
            ...defaultBookProps,
            added: 'read',
        };

        const book = shallow(<Book {...bookProps} />);

        expect(book.find(BookCondition).dive().text()).toEqual('read');
    });

    it('should call onChangeShelf prop when ShelfChanger trigger onChangeShelf function', () => {
        const bookProps = {
            ...defaultBookProps,
            onChangeShelf: jest.fn(),
        };

        const book = mount(<Book {...bookProps} />);

        book.find('ShelfChanger').instance().props.onChangeShelf();

        expect(bookProps.onChangeShelf).toHaveBeenCalled();
    });
});
