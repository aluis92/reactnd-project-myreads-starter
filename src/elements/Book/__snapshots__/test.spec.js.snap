// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book component should render without crashing 1`] = `
<li>
  <styled.div
    className="book"
  >
    <styled.div>
      <styled.div
        style={
          Object {
            "backgroundImage": "url(undefined)",
          }
        }
      />
      <ShelfChanger
        defaultValue={null}
        onChangeShelf={[Function]}
      />
    </styled.div>
    <styled.h3 />
    <styled.p />
  </styled.div>
</li>
`;
